### PR TITLE
[REACTOS] Sync' __cxa_pure_virtual() definitions

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Consistency, self-documentation, tiny compilation optimization

## Proposed changes

- Sync' __cxa_pure_virtual() definitions
- and compile them when needed only.

### CODE REVIEW

File: dll/win32/shdocvw/mrulist.cpp [hunk 1]
```diff
@@ -32,11 +33,13 @@ class CMruBase;
             class CMruPidlList;
 class CMruClassFactory;
 
-extern "C" void __cxa_pure_virtual(void)
+#ifdef __GNUC__ // GCC and Clang.
+extern "C" void __cxa_pure_virtual()
```
Comment (HBelusca) [inline dll/win32/shdocvw/mrulist.cpp:37]: What's happening with MSVC builds then?
Do we also know why those things were needed only in these three C++ modules, and not all the others?

Comment (SergeGautherie) [reply dll/win32/shdocvw/mrulist.cpp:37]: > What's happening with MSVC builds then?

Per a web search, it looks like `__cxa_pure_virtual()` is GCC (and Clang) specific.
Compilation with MSVC (and Clang-cl) succeed, which would seem to confirm that..

>  Do we also know why those things were needed only in these three C++ modules, and not all the others?

I have no idea: I am simply sync'ing (and narrowing) existing code.

You might ask authors:

mrulist.cpp: 2023-08-31
https://github.com/reactos/reactos/commit/58770200e75bd8f0c44485b0281fbd97fa799406#diff-139e6c11bf066b06e72abbd5e181a27e0bdfeadd426c8ece544a94096a8f8f61

purecall.cpp : r43026
https://github.com/reactos/reactos/commit/ad2cf72549b9e95e537ff9291cd8187ae4948d38#diff-2986c8a086e3bd8f84009939250e7a788e7c9b852b7137aa8d41309df1f4854d

cunknown.cpp : r58641
https://github.com/reactos/reactos/commit/a08dba228ca0bf3a63c3e27ae59325b51af780cd#diff-b53cfcab95e7a76d75b6887940184462cebe6f8ff08cf4f0109fea547b969dbc

reactos_special.cpp: 2020-11-02
https://github.com/reactos/reactos/commit/1f377076d783d597f33e720498ef29505973c680#diff-062c49329af0b4110974cfa6b07a88b4ae329bb1b1789fab77482e29b16f54bd

**Review by learn-more — CHANGES_REQUESTED** [review]
I am not in favor of micro-optimizing hacks for compilers like this.

**Comment by learn-more** [comment]
> > requested changes
> 
> Which changes would you like, more precisely?

That is the only option I have other than closing it immediately.
if other devs agree with me then I propose we close this, otherwise my feedback can be discarded and the PR can continue.

File: dll/win32/shdocvw/mrulist.cpp (hunk 1 above)
Comment (SergeGautherie) [inline dll/win32/shdocvw/mrulist.cpp:37]: Since then, this very occurrence was moved on 960a305, then removed/replaced on #7467 678aa63.

**Comment by SergeGautherie** [comment]
Please, reopen. This PR would also sync with #7467.

**Pull request closed without merging**